File: src/api/FleetManager/FleetManager.Domain/Orders/Order.cs
namespace FleetManager.Domain.Orders
{
    public class Order
    {
        public Guid Id { get; private set; }
        public Guid ContractorId { get; private set; }
        public Guid OriginId { get; private set; }
        public Guid DestinationId { get; private set; }
        public DateTime PickupDate { get; private set; }
        public DateTime DeliveryDate { get; private set; }
        public OrderStatus Status { get; private set; }

        internal Order(
            Guid contractorId,
            Guid originId,
            Guid destinationId,
            DateTime pickupDate,
            DateTime deliveryDate)
        {
            ContractorId = contractorId;
            OriginId = originId;
            DestinationId = destinationId;
            PickupDate = pickupDate;
            DeliveryDate = deliveryDate;
            Status = OrderStatus.Created;
        }
    }
}

File: src/client/web/FleetManager/src/app/features/orders/order-management/models/order.ts
export class Order{
  public id: string;
  public contractorName: string;
  public originLocation: string;
  public destinationLocation: string;
  public pickupDate: Date;
  public deliveryDate: Date;
  public status: string;

  constructor(id: string, contractorName: string, originLocation: string, destinationLocation: string, pickupDate: Date, deliveryDate: Date, status: string) {
    this.id = id;
    this.contractorName = contractorName;
    this.originLocation = originLocation;
    this.destinationLocation = destinationLocation;
    this.pickupDate = pickupDate;
    this.deliveryDate = deliveryDate;
    this.status = status;
  }
}
